modules/nlu/src/backend/prediction-handler.ts

```diff
@@ -68,7 +71,16 @@ export class PredictionHandler {
       this.modelsByLang[lang] = model.hash
       await this.engine.loadModel(model, model.hash)
     }
-    return this.engine.predict(textInput, includedContexts, this.modelsByLang[lang])
+
+    const spellChecked = await this.engine.spellCheck(textInput, this.modelsByLang[lang])
+    if (spellChecked !== textInput) {
+      const originalOutput = await this.engine.predict(textInput, includedContexts, this.modelsByLang[lang])
+      const spellCheckedOutput = await this.engine.predict(spellChecked, includedContexts, this.modelsByLang[lang])
+      const merged = mergeSpellChecked(originalOutput as PredictOutput, spellCheckedOutput as PredictOutput)
+      return { ...merged, spellChecked }
```

Comment:
Reading this fast I got confused why you we merge what's been merged but now i get that spellChecked is spellcheckedInput.

can we rename that or you think this would be breaking ?

Reply:
😕  not sure what you mean... We still have to put the result of spell checking inside the `event.nlu`object?

From a `nlu-core`users perspective, `spellChecked`is an input, bot for the botpress user perspective, `spellChecked`is an output...